test/duckula/avro/schema_test.clj

```diff
@@ -1,35 +1,79 @@
 (ns duckula.avro.schema-test
   (:require
-    [abracad.avro.codec :as avro.codec]
-    [cheshire.core :as json]
-    [clojure.test :refer [is deftest testing]]
-    [duckula.avro]
-    [duckula.avro.schema :as avro.schema]
-    [schema.core]))
-
+   [abracad.avro.codec :as avro.codec]
+   [cheshire.core :as json]
+   [clojure.test :refer [is deftest testing]]
+   [duckula.avro]
+   [duckula.avro.schema :as avro.schema]
+   [schema.core :as s]))
 
 (def sample-avro-schema
   (duckula.avro/load-schemas "schema/endpoint/search/test/Response.avsc"))
 
+(defn build-avro-schema [shape]
+  (avro.codec/parse-schema*
+   (json/generate-string
+    shape)))
 
 (deftest generates-a-prismatic-schema-from-avro
   (let [payload {:status "error"
                  :items [{:content "foo" :id 1}
                          {:content "bar" :id 2}]
                  :message "foo"}]
     (is (= payload
-           (schema.core/validate
-             (avro.schema/->map {:avro-schema sample-avro-schema})
-             payload)))))
-
+           (s/validate
+            (avro.schema/->map {:avro-schema sample-avro-schema})
+            payload)))))
 
 (deftest converts-avro-to-primastic-schema
-  (let [avro-schema (avro.codec/parse-schema*
-                      (json/generate-string
-                        {:name "Foo"
-                         :doc "a foo!"
-                         :type "record"
-                         :fields [{:name "bar" :type "string" :doc "bah"}]}))
-        expected-schema {:bar avro.schema/string}]
-    (is (= expected-schema
-           (avro.schema/->map {:avro-schema  avro-schema})))))
+  (testing "simple case"
+    (let [avro-schema (build-avro-schema
+                       {:name "Foo"
+                        :doc "a foo!"
+                        :type "record"
+                        :fields [{:name "foo_bar" :type "string" :doc "bah"}]})
+          expected-schema {:foo_bar avro.schema/string}]
+      (is (= expected-schema
+             (avro.schema/->map {:avro-schema avro-schema})))))
+
+  (testing "name mangling :-("
```

Comment:
So this was always confusing for me. Can you verify 
1.) scheme always needs to be define with underscore in keys (so `[{:name "foo-bar" :type "string" :doc "bah"}]` is not allowed)
2.) frontend can only send payload `{"foo_bar":"test"}` , `{"foo-bar":"test"}` would fail. Setting `mangle-names?` to true just means we will convert the data to `{:foo-bar "test"}` before passing to handler fn? 
3.) We need some guidelines when to use `mangle-names?`

Reply:
It **can** be alowed, if you define config like this:


```clojure
{:name "api"
 :endpoints {
   "/foo" { :handler ... }
  }
}
```

Then request and response **require** using `foo_bar`, for map keys and enums.

If you do this:


```clojure
{:name "api"
 :mangle-names? true  ;; or :kebab-case-names? true
 :endpoints {
   "/foo" { :handler ... }
  }
}
```

then request and response **require** using `foo-bar` when sending data to and from the endpoint.
As for guidance - I think the default makes more sense since our use is mostly around Typescript frontend talking to a set of APIs, service to service calls are not that common (yet) and Clojure is just fine dealing with either `kebab-case` and `snake_case` and with middlewares you can fix that in code, rather than the API layer.
As I mentioned above, in retrospect we should have never made this into an option, and need to review internally how this setting is used (I'd assume we don't use `mangle-names` at all)